library/Vanilla/Exception/ConfigurationException.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * @author Alexandre (DaazKu) Chouinard <alexandre.c@vanillaforums.com>
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ */
+
+namespace Vanilla\Exception;
+
+use Garden\Web\Exception\ForbiddenException;
+
+/**
+ * An exception tha represents a configuration test failing.
+ */
+class ConfigurationException extends ForbiddenException {
+    /**
+     * Construct a {@link ConfigurationException} object.
+     *
+     * @param string The configuration that failed the test.
+     * @param array $context Additional information for the error.
+     */
+    public function __construct($configuration, array $context = []) {
+        $context['configuration'] = $configuration;
+
+        $msg = sprintf('The %s configuration value disallow you to do that.', $configuration);
```

Comment:
Some configurations must have a specific value such as registration method. I imagine this exception would be nice if you could

1. Specify a required value which defaults to true.
2. Display an error message that nicely works for both cases.
3. Add a t() wrap so that we can make adjustments to non-friendly error messages if we want to.

```php
if ($value === true) {
    $msg = t("The $configuration config is required to access this.");
} else {
   $message = sprintf(t("The $configuration config must be \"%s\" to access this."), $this->translateValue($value));
}
```